arroyo/processing/strategies/run_task.py

```diff
@@ -49,16 +50,19 @@ class RunTask(ProcessingStrategy[TPayload], Generic[TPayload, TResult]):
 
     def __init__(
         self,
-        function: Callable[[Message[TPayload]], TResult],
+        function: Callable[[BaseValue[TPayload]], TResult],
         next_step: ProcessingStrategy[TResult],
     ) -> None:
         self.__function = function
         self.__next_step = next_step
 
     def submit(self, message: Message[TPayload]) -> None:
-        result = self.__function(message)
-        value = message.value.replace(result)
-        self.__next_step.submit(Message(value))
+        if isinstance(message.value.payload, FilteredPayload):
```

Comment:
So that's what I had at the very beginning, I changed all strategies' baseclasses to read `ProcessingStrategy[Union[...]]`, Later it was suggested that we attempt to hide the filtered-ness in message, so the type of `message.payload` for a `message: Message[T]` is `Union[T, FilteredPayload]`.

The compiler does warn you about that. If this case isn't handled, pyright/mypy will warn you about `message.value.payload`/`message.payload` potentially being `FilteredPayload`.

Reply:
Makes sense, I think i missed this earlier.